lib/util/cmcd_manager.js

```diff
@@ -480,6 +480,12 @@ shaka.util.CmcdManager = class {
         result = key;
       } else if (type === 'symbol') {
         result = `${key}=${value.description}`;
+      } else if (value == null) {
+        // On some platforms (Xbox), `key=${undefined}` returns
+        // 'key=undefined', whereas on others, it returns 'key='.  For
+        // consistency in output and in tests, check for null & undefined
+        // explicitly.
+        result = `${key}=`;
```

Comment:
Neither `key=` and `key=undefined` are allowed values in CMCD. `null` and `undefined` values should be skipped entirely. This is why the `isValid` check occurs on line 453 which protects against `undefined|null` values. How exactly is player getting into this state?

Reply:
I think the correct fix would be to find out why `isValid` is returning `true` when passed an `undefined` value on Xbox.

Reply:
Actually, I think this whole PR is wrong.  I messed up somewhere in analyzing this and verifying the fix.  The broken test is still failing.  Sorry for the noise.